rust_src/src/ng_async.rs

```diff
@@ -239,14 +368,59 @@ pub async fn async_echo(s: String) -> String {
     s
 }
 
+#[async_stream]
+pub async fn async_data_echo(e: UserData) -> UserData {
+    e
+}
+
+fn internal_send_message(
+    pipe: &mut EmacsPipe,
+    message: LispObject,
+    option: PipeDataOption,
+) -> bool {
+    match option {
+        PipeDataOption::STRING => {
+            if !is_string(message) {
+                wrong_type!(Qstringp, message);
+            }
+
+            let sdata = unsafe { SDATA(message) };
+            let ssize = unsafe { SBYTES(message) };
+            let sslice = unsafe { slice::from_raw_parts(sdata as *const u8, ssize as usize) };
+            let contents = String::from_utf8_lossy(sslice);
```

Comment:
Maybe we should add a method on `LispObject` that directly converts it to a `String`.

Reply:
Removed all usages of SDATA and SBYTES directly. Now we are using code from multibyte.rs